### PR TITLE
fix: use dylib extension instead of jnilib for MacOS library names

### PR DESCRIPTION
**Pull request by kkriske**
copy from https://github.com/xerial/sqlite-jdbc/pull/922#issuecomment-1595180965:
The libraries for MacOS are packaged as a .jnilib file. This seems legacy from pre java 7 (https://bugs.openjdk.org/browse/JDK-7134701). The default naming now seems to be .dylib.

### CODE REVIEW

File: src/main/resources/org/sqlite/native/Windows/armv7/sqlitejdbc.dll
Comment (gotson) [inline src/main/resources/org/sqlite/native/Windows/armv7/sqlitejdbc.dll:1]: we probably don't need that one

File: src/main/resources/org/sqlite/native/Windows/x86/sqlitejdbc.dll
Comment (gotson) [inline src/main/resources/org/sqlite/native/Windows/x86/sqlitejdbc.dll:1]: we probably don't need that one

File: src/main/resources/org/sqlite/native/Windows/x86_64/sqlitejdbc.dll
Comment (gotson) [inline src/main/resources/org/sqlite/native/Windows/x86_64/sqlitejdbc.dll:1]: we probably don't need that one

File: src/main/resources/org/sqlite/native/Windows/aarch64/sqlitejdbc.dll
Comment (gotson) [inline src/main/resources/org/sqlite/native/Windows/aarch64/sqlitejdbc.dll:1]: we probably don't need that one

File: src/main/resources/org/sqlite/native/Mac/aarch64/libsqlitejdbc.dylib
Comment (gotson) [inline src/main/resources/org/sqlite/native/Mac/aarch64/libsqlitejdbc.dylib:1]: was that regenerated or just renamed ? file size has changed

File: src/main/resources/org/sqlite/native/Mac/x86_64/libsqlitejdbc.dylib
Comment (gotson) [inline src/main/resources/org/sqlite/native/Mac/x86_64/libsqlitejdbc.dylib:1]: was that regenerated or just renamed ? file size has changed

**Review by gotson — CHANGES_REQUESTED** [review]

File: src/main/resources/org/sqlite/native/Mac/aarch64/libsqlitejdbc.dylib
Comment (kkriske) [reply src/main/resources/org/sqlite/native/Mac/aarch64/libsqlitejdbc.dylib:1]: I renamed them initially, and then ran the CI to validate if they are still created correctly.
I don't know why the file size is different.

File: src/main/resources/org/sqlite/native/Windows/armv7/sqlitejdbc.dll
Comment (kkriske) [reply src/main/resources/org/sqlite/native/Windows/armv7/sqlitejdbc.dll:1]: The change does nothing indeed, but windows is unable to create the exact same binary twice due to embedded metadata like creation timestamps. I can revert the changes to the dlls if you want.

Comment (gotson) [reply src/main/resources/org/sqlite/native/Windows/armv7/sqlitejdbc.dll:1]: > I can revert the changes to the dlls if you want.

that's what i was implying. This PR should not modify those files.

File: src/main/resources/org/sqlite/native/Mac/aarch64/libsqlitejdbc.dylib
Comment (gotson) [reply src/main/resources/org/sqlite/native/Mac/aarch64/libsqlitejdbc.dylib:1]: can we keep the original file and just rename them in that case?

Comment (kkriske) [reply src/main/resources/org/sqlite/native/Mac/aarch64/libsqlitejdbc.dylib:1]: I reverted the CI run, that should do it.

**Pull request merged**